lib/storage/metadata/mongoclient/readStream.js

```diff
@@ -85,7 +85,8 @@ class MongoReadStream extends Readable {
             Object.assign(query, searchOptions);
         }
 
-        this._cursor = c.find(query).sort({
+        const projection = { 'value.location': 0 };
+        this._cursor = c.find(query, { projection }).sort({
```

Comment:
LGTM, but would be best to add a functional test case : to verify we do indeed drop this attribute form listing
(maybe in https://github.com/scality/Arsenal/blob/development/8.1/tests/functional/metadata/mongodb/listObject.spec.js ?)